python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -374,12 +375,21 @@ def new_version_storage_id(
             else None
         )
 
-        next_version_record = self.next_version_record(
+        latest_version_record = self.get_observation_record(
```

Comment:
note: previously, this would return the next asset observation after the cursor with a different version. So if you had (storage_id, version) of:

`[(1, a), (2, b), (3, b), (4, b), (5, c), (6, d), (7, d)]`

Then `f(2)` would previously return `5` (the next record with a different version), and now will return `7` (the latest record id, if it has a different version).

If you added `(8, b)` to the end of the list, then `f(2)` would return `None`, as the version is the same as it was at `2`.